sql/src/main/java/io/crate/planner/statement/SetLicensePlan.java

```diff
@@ -0,0 +1,93 @@
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.statement;
+
+import io.crate.analyze.SetLicenseAnalyzedStatement;
+import io.crate.analyze.SymbolEvaluator;
+import io.crate.data.Row;
+import io.crate.data.Row1;
+import io.crate.data.RowConsumer;
+import io.crate.execution.support.OneRowActionListener;
+import io.crate.license.LicenseMetaData;
+import io.crate.license.SetLicenseRequest;
+import io.crate.planner.DependencyCarrier;
+import io.crate.planner.Plan;
+import io.crate.planner.PlannerContext;
+import io.crate.planner.operators.SubQueryResults;
+import io.crate.types.DataTypes;
+import org.apache.lucene.util.BytesRef;
+
+import java.util.Locale;
+
+public class SetLicensePlan implements Plan {
+
+    private final SetLicenseAnalyzedStatement stmt;
+
+    public SetLicensePlan(final SetLicenseAnalyzedStatement stmt) {
+        this.stmt = stmt;
+    }
+
+    @Override
+    public StatementType type() {
+        return StatementType.MANAGEMENT;
+    }
+
+    @Override
+    public void execute(DependencyCarrier executor,
+                        PlannerContext plannerContext,
+                        RowConsumer consumer,
+                        Row params,
+                        SubQueryResults subQueryResults) {
+
+        Long expiryDateMs = null;
+        BytesRef issueTo = null;
+        BytesRef signature = null;
+        try {
+            expiryDateMs = DataTypes.TIMESTAMP.value(
+                SymbolEvaluator.evaluate(plannerContext.functions(),
+                    stmt.expirationDateSymbol(),
+                    params, subQueryResults));
+
+            issueTo = DataTypes.STRING.value(
+                SymbolEvaluator.evaluate(plannerContext.functions(),
+                    stmt.issuedToSymbol(),
+                    params, subQueryResults));
+
+            signature = DataTypes.STRING.value(
+                SymbolEvaluator.evaluate(plannerContext.functions(),
+                    stmt.signatureSymbol(),
+                    params, subQueryResults));
+        } catch (ClassCastException e) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
```

Comment:
I think it would be nice to have the `try/catch` *per* setting so that we could include in the error message more detail about **which** setting is invalid. (Might want to extract the `try/catch` into a static method to avoid the boilerplate from piling up)